lib/turbostreamer.rb

```diff
@@ -280,22 +280,17 @@ def child!(value = BLANK, *args, &block)
     else
       object!{ extract!(value, *args) }
     end
-    
+
   end
-  
+
   # Encodes the current builder as JSON.
   def target!
     @encoder.flush
-    
-    if @encoder.output.is_a?(::StringIO)
-      @encoder.output.string
-    else
-      @encoder.output
-    end
+    @encoder.to_output
   end
```

Comment:
Think we should keep this here instead of using `to_output`, I think this function will end up being the same on all the encoders so we can just do it here. I'm guessing it was probably added to remove the newline for test on the OJ encoder, but I'll just modify the test to accept surrounding whitespace. If you think it could be different on encoders then I'm open to adding it.

Maybe in the future just commit the changes to the file instead of the changes + removing trailing whitespace. I'm not on either boat but if we want to remove trailing whitespace we can do that in a separate commit to make the diffs easier to read.

Reply:
Right now it's only for removing newline char
But I am not sure what do you want for an encoder's interface

I got this list from reading the code:
- `to_output`: Return output, string form only? (seems not)
- `flush`: Flush buffer to output
- `key`: Add a key to object
- `value`: Add a value to object (after calling `key`)
- `capture`: capture output (sometimes including keys) and return as string (without surrounding brackets and commas)
- `inject`: push string as new value to json
- `map_open`: Open an object and enter it on next push
- `map_close`: Close an object and go one level upper for next push
- `array_open`: Open an array and enter it on next push
- `map_close`: Close an array and go one level upper for next push